src/object.js

```diff
@@ -146,11 +160,43 @@ ReflectionObject.prototype.onRemove = function onRemove(parent) {
 ReflectionObject.prototype.resolve = function resolve() {
     if (this.resolved)
         return this;
-    if (this.root instanceof Root)
-        this.resolved = true; // only if part of a root
+    if (this.root instanceof Root || this.parent) {
+        this._resolveFeatures();
+        this.resolved = true;
+    }
     return this;
 };
 
+/**
+ * Resolves child features from parent features
+ * @returns {undefined}
+ */
+ReflectionObject.prototype._resolveFeatures = function _resolveFeatures() {
+    var defaults = {};
+
+    if (this.root.getOption("syntax") === "proto2") {
+        defaults = Object.assign({}, proto2Defaults);
+    } else if (this.root.getOption("syntax") === "proto3") {
+        defaults = Object.assign({}, proto3Defaults);
+    } else if (this.root.getOption("edition") === "2023") {
+        defaults = Object.assign({}, editions2023Defaults);
+    }
+
+    if (this.parent) {
+        // This is an annoying workaround since we can't use the spread operator
+        // (Breaks the bundler and eslint)
+        // If we don't create a shallow copy, we end up also altering the parent's
+        // features
+        var parentFeaturesMerged = Object.assign(defaults, this.parent._proto_features);
+        this._features = Object.assign(parentFeaturesMerged, this._proto_features || {});
+        this.parent._resolveFeatures();
+    } else {
+        this._features = Object.assign(defaults, this._proto_features || {});
+    }
+    this._proto_features = this._features;
```

Comment:
O, my earlier comments may be related to some confusion here.  My understanding is that _proto_features contains the original specification in the proto file.  Why is it being replaced with duplicate data here?

Reply:
Fixed per our conversation.